bot/utils/safety_api.py

```diff
@@ -0,0 +1,41 @@
+import aiohttp
+from urllib import parse
+from loguru import logger
+
+from bot.constants import SAFETY_API_ENDPOINT, SAFETY_IDS
+
+
+def key_sort(article_id: int):
+    try:
+        category_index = SAFETY_IDS.index(article_id)
+    except ValueError:
+        category_index = 2000  # Number which shall never be reached in articles
+    return category_index
+
+
+async def get_api(query_string: str = None, category: int = None):
+    url_params = {}
+    if query_string:
+        url_params["query"] = f"{{{query_string}}}"
+
+    if category:
+        url_params["section"] = category
+
+    request_url = f"{SAFETY_API_ENDPOINT}?{parse.urlencode(url_params)}"
+
+    results_parse = []
+    try:
+        async with aiohttp.ClientSession() as sess:
+            response = await sess.get(request_url)
+            response.raise_for_status()
+            data = await response.json()
+
+            results = data["results"]
+            for result in results:
+                result["url"] = f"https://discord.com/safety/{result['id']}"
+
+            results_parse = sorted(results, key=lambda x: key_sort(x["id"]))
+
+    except Exception:
+        logger.exception("Error fetching information from Safety")
```

Comment:
Please log the error too

Reply:
@vladfrangu `loguru` will automatically capture/append exception data when using `log.exception()` as an `ERROR` class.

Reply:
I will also note (not as a review, but just as a design note) - we are eating the exception (very broadly) here and not allowing it to propagate up the chain. If the API encounters an error, there will be no way for the end user to know other than "wait, why did nothing return from my search for something I know exists?"

I don't know how we want to handle underlying errors or if we want to even expose them to end users. As such, I can't offer any sort of advice to fix this -- I just wanted to put this on the record as food-for-thought down the line.

Reply:
Yeah, in this case we are eating it broadly, but the only reason that we would have an error thrown here as far as I am aware is a failure to fetch from the Zendesk API, a failure to jsonify the response, or the Zendesk API changing completely and removing the `results` key in favour of another field. I personally don't see an issue here, but if others do we could make erroring more specific in this case.

Reply:
Added user feedback from error throwing with a `Couldn't load search results from the Safety Portal, please contact the devs` if error thrown, and changed the logging message to `Error in Safety API Request`